projects/adrv9001/src/hal/no_os_platform.c

```diff
@@ -0,0 +1,671 @@
+	gip_gpio_ssi_sync.extra = &gip_extra;
+	gip_gpio_ssi_sync.platform_ops = &xil_gpio_platform_ops;
+	ret = gpio_get(&phal->gpio_ssi_sync, &gip_gpio_ssi_sync);
+	if (ret < 0)
+		return ret;
+
+	ret = gpio_direction_output(phal->gpio_ssi_sync, GPIO_LOW);
+	if (ret)
+		return ret;
+#endif
+	struct spi_init_param sip = {
+		.max_speed_hz = 20000000,
+		.mode = SPI_MODE_0,
+		.chip_select = SPI_CS,
+		.platform_ops = &xil_platform_ops,
+		.extra = &sip_extra
+	};
+	ret = spi_init(&phal->spi, &sip);
+	if (ret)
+		return ret;
+
+	return ADI_HAL_OK;
+}
+
+/**
+ * @brief Gracefully shuts down the the hardware closing any open resources
+ *        such as log files, I2C, SPI, GPIO drivers, timer resources, etc.
+ *
+ * @param devHalCfg Pointer to device instance specific platform settings
+ *
+ * @retval ADI_HAL_OK Function completed successfully, no action required
+ * @retval ADI_HAL_NULL_PTR The function has been called with a null pointer
+ */
+int32_t no_os_HwClose(void *devHalCfg)
+{
+	int32_t ret;
+	adi_hal_Cfg_t *phal = (adi_hal_Cfg_t *)devHalCfg;
+	ret = gpio_remove(phal->gpio_reset_n);
+	if (ret)
+		return ret;
+
+	ret = spi_remove(phal->spi);
+	if (ret)
+		return ret;
+
+	return ADI_HAL_OK;
+}
+
+/**
+ * @brief This function control a BBIC GPIO pin that connects to the reset pin
+ *        of each device.
+ *
+ *  This function is called by each device API giving access to the Reset pin
+ *  connected to each device.
+ *
+ * @param devHalCfg Pointer to device instance specific platform settings
+ * @param pinLevel The desired pin logic level 0=low, 1=high to set the GPIO pin to.
+ *
+ * @retval ADI_HAL_OK Function completed successfully, no action required
+ * @retval ADI_HAL_NULL_PTR The function has been called with a null pointer
+ */
+int32_t no_os_HwReset(void *devHalCfg, uint8_t pinLevel)
+{
+	adi_hal_Cfg_t  *phal = (adi_hal_Cfg_t *)devHalCfg;
+
+	if (!devHalCfg)
+		return ADI_HAL_NULL_PTR;
+
+	gpio_set_value(phal->gpio_reset_n, pinLevel);
+
+	return ADI_HAL_OK;
+}
+
+/**
+ * @brief Write an array of 8-bit data to a SPI device
+ *
+ * The function will write numTxBytes number of bytes to the SPI device
+ * selected in the devHalCfg structure.
+ *
+ * @param devHalCfg Pointer to device instance specific platform settings
+ * @param txData Pointer to byte array txData buffer that has numTxBytes number of bytes
+ * @param numTxBytes The length of txData array
+ *
+ * @retval ADI_HAL_OK function completed successfully, no action required
+ * @retval ADI_HAL_NULL_PTR the function has been called with a null pointer
+ * @retval ADI_HAL_SPI_FAIL the data was not written successfully
+ */
+int32_t no_os_SpiWrite(void *devHalCfg, const uint8_t txData[],
+		       uint32_t numTxBytes)
+{
+	static const int32_t MAX_SIZE = 4096;
+	uint32_t toWrite = 0;
+	int32_t remaining = numTxBytes;
+	int32_t halError = (int32_t)ADI_HAL_OK;
+	adi_hal_Cfg_t *halCfg = NULL;
+
+	if (devHalCfg == NULL) {
+		halError = (int32_t)ADI_HAL_NULL_PTR;
+		return halError;
```

Comment:
`halError` doesn't seem to have a point here. We can just return the error value directly. Same with the last statement.

Reply:
true, i removed it